src/MagnumPlugins/WebPImporter/Test/WebPImporterTest.cpp

```diff
@@ -0,0 +1,304 @@
+
+    addInstancedTests({&WebPImporterTest::invalid},
+        Containers::arraySize(InvalidData));
+
+    addInstancedTests({&WebPImporterTest::rgb},
+        Containers::arraySize(RgbData));
+
+    addInstancedTests({&WebPImporterTest::rgba},
+        Containers::arraySize(RgbaData));
+
+    addInstancedTests({&WebPImporterTest::openMemory},
+        Containers::arraySize(OpenMemoryData));
+
+    addTests({&WebPImporterTest::openTwice,
+              &WebPImporterTest::importTwice});
+
+    /* Load the plugin directly from the build tree. Otherwise it's static and
+       already loaded. */
+    #ifdef WEBPIMPORTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT_OUTPUT(_manager.load(WEBPIMPORTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+}
+
+void WebPImporterTest::empty() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("WebPImporter");
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    char a{};
+    /* Explicitly checking non-null but empty view */
+    CORRADE_VERIFY(!importer->openData({&a, 0}));
+    CORRADE_COMPARE(out.str(), "Trade::WebPImporter::openData(): the file is empty\n");
+}
+
+void WebPImporterTest::invalid() {
+    auto&& data = InvalidData[testCaseInstanceId()];
+    setTestCaseDescription(data.name);
+
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("WebPImporter");
+
+    Containers::Optional<Containers::Array<char>> in = Utility::Path::read(Utility::Path::join(WEBPIMPORTER_TEST_DIR, data.filename));
+    CORRADE_VERIFY(in);
+
+    /* The open does just a memory copy, so it doesn't fail */
+    CORRADE_VERIFY(importer->openData(data.size ? in->prefix(*data.size) : *in));
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    CORRADE_VERIFY(!importer->image2D(0));
+    CORRADE_COMPARE(out.str(), Utility::formatString("Trade::WebPImporter::image2D(): {}", data.error));
+}
+
+void WebPImporterTest::rgb() {
+    auto&& data = RgbData[testCaseInstanceId()];
+    setTestCaseDescription(data.name);
+
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("WebPImporter");
+    CORRADE_VERIFY(importer->openFile(Utility::Path::join(WEBPIMPORTER_TEST_DIR, data.filename)));
+
+    Containers::Optional<Trade::ImageData2D> image = importer->image2D(0);
+    CORRADE_VERIFY(image);
+    CORRADE_COMPARE(image->size(), Vector2i(3, 3));
+    CORRADE_COMPARE(image->format(), PixelFormat::RGB8Unorm);
+    CORRADE_COMPARE(image->storage().alignment(), 4);
+
+    /* The image has four-byte aligned rows, clear the padding to deterministic
+       values */
+    CORRADE_COMPARE(image->data().size(), 36);
+    image->mutableData()[9] = image->mutableData()[10] =
+        image->mutableData()[11] = image->mutableData()[21] =
+            image->mutableData()[22] = image->mutableData()[23] =
+                image->mutableData()[33] = image->mutableData()[34] =
+                        image->mutableData()[35] = 0;
```

Comment:
This block shouldn't be needed anymore with `CompareImage`, it ignores the padding. 

(It's still needed in case of `openMemory()` tho, keep it there.)